hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -890,32 +924,37 @@ int SaraNcpClient::initReady() {
     if (conf_.ncpIdentifier() != PLATFORM_NCP_SARA_R410) {
         // Change the baudrate to 921600
         CHECK(changeBaudRate(UBLOX_NCP_RUNTIME_SERIAL_BAUDRATE_U2));
-        // Check that the modem is responsive at the new baudrate
-        skipAll(serial_.get(), 1000);
-        CHECK(waitAtResponse(10000));
-    }
+    } else {
+        int fwVersion = getAppFirmwareVersion();
+        if (fwVersion > 0) {
+            // L0.0.00.00.05.06,A.02.00 has a memory issue
+            memoryIssuePresent_ = (fwVersion == UBLOX_NCP_R4_APP_FW_VERSION_MEMORY_LEAK_ISSUE);
+            // There is a set of other revisions which do not have hardware flow control
+            if (!(fwVersion >= UBLOX_NCP_R4_APP_FW_VERSION_NO_HW_FLOW_CONTROL_MIN &&
+                    fwVersion <= UBLOX_NCP_R4_APP_FW_VERSION_NO_HW_FLOW_CONTROL_MAX)) {
+                // Change the baudrate to 460800
+                // NOTE: ignoring AT errors just in case to accommodate for some revisions
+                // potentially not supporting anything other than 115200
+                int r = changeBaudRate(UBLOX_NCP_RUNTIME_SERIAL_BAUDRATE_R4);
+                if (r != SYSTEM_ERROR_AT_NOT_OK) {
```

Comment:
May need to check for AT here, before we tend to set `AT+IFC` in the next lines
```
    // Check that the modem is responsive at the new baudrate
    skipAll(serial_.get(), 1000);
    CHECK(waitAtResponse(10000));
```

Reply:
Yeah, you are right, let's do that.

Reply:
Simply moved `AT+IFC=2,2` below that check.